### PR TITLE
Libvirt Driver for runv

### PR DESCRIPTION
**Pull request by gao-feng**
Edit /etc/hyper/config, change or add the `Hypervisor:libvirt`.

You should change the file mode of `kernel`, `initrd` to fix the Permission denied problem. or change the /etc/libvirt/qemu.conf to make qemu run as root user.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -6,7 +6,9 @@ matrix:
```
Comment (feiskyer) [inline .travis.yml]: Add these two lines at the head of .travis.yml and try again to compile correctly: 

```
sudo: required
dist: trusty
```

File: .travis.yml [hunk 2]
```diff
@@ -6,7 +6,9 @@ matrix:
 
 before_install:
   - sudo apt-get update -qq
-  - sudo apt-get install -y autoconf libdevmapper-dev -qq
+  - sudo apt-get install -y autoconf libdevmapper-dev libvirt-dev -qq
+  - wget http://security.ubuntu.com/ubuntu/pool/main/libv/libvirt/libvirt-dev_1.2.21-2ubuntu2_amd64.deb
+  - sudo dpkg -i --force-all libvirt-dev_1.2.21-2ubuntu2_amd64.deb
```
Comment (feiskyer) [inline .travis.yml]: Line #10 and #11 should be removed on trusty.

File: .travis.yml (hunk 1 above)
Comment (gao-feng) [inline .travis.yml]: Thanks! This save the time to compile libvirt manually

**Comment by feiskyer** [comment]
Can't start pod in my env, and `hyper list` incorrectly shows the pod is running:

```sh
[root@k8s ~]# hyper run -p hyper.json
POD id is pod-klvgQuUkGY
hyper ERROR: Error from daemon's response: VM response data is nil
[root@k8s ~]# hyper list
POD ID              POD Name            VM name             Status
pod-klvgQuUkGY      jadac916s1          vm-IIcJjTVBuK       running
[root@k8s ~]# ps -ef | grep qemu
root     24273 18135  0 06:55 pts/3    00:00:00 grep --color=auto qemu
```

logs:

```
I1221 06:54:38.884335   23657 pod.go:615] configure container log to /var/run/hyper/Pods/pod-klvgQuUkGY/593e349accc4e842e3a8f9a1dc3550a725d559ee8d19a6727185db04b7c4bb54-json.log
I1221 06:54:38.884512   23657 pod.go:621] configured logger for pod-klvgQuUkGY/593e349accc4e842e3a8f9a1dc3550a725d559ee8d19a6727185db04b7c4bb54 (/centos)
I1221 06:54:38.884594   23657 vm.go:200] hyperHandlePodEvent pod pod-klvgQuUkGY, vm vm-IIcJjTVBuK
I1221 06:54:38.884636   23657 vm.go:290] return response &{ -1 Start pod failed <nil> <nil>}
E1221 06:54:38.884656   23657 pod.go:101] VM response data is nil
I1221 06:54:38.884661   23657 pod.go:102] unlock PodList
I1221 06:54:38.884672   23657 job.go:84] -job podStart(pod-klvgQuUkGY, vm-IIcJjTVBuK, ) ERR: VM response data is nil
E1221 06:54:38.884683   23657 server.go:1040] Handler for POST /pod/start returned error: VM response data is nil
E1221 06:54:38.884693   23657 server.go:132] HTTP Error: statusCode=500 VM response data is nil
E1221 06:54:48.742566   23657 tty.go:124] Cannot connect to tty socket dial unix /var/run/hyper/vm-IIcJjTVBuK/tty.sock: no such file or directory
E1221 06:54:48.742641   23657 init_comm.go:29] failed to connected to /var/run/hyper/vm-IIcJjTVBuK/console.sock dial unix /var/run/hyper/vm-IIcJjTVBuK/console.sock: no such file or directory
E1221 06:54:48.742674   23657 init_comm.go:106] Cannot connect to hyper socket dial unix /var/run/hyper/vm-IIcJjTVBuK/hyper.sock: no such file or directory
```

**Comment by gao-feng** [comment]
No useful log, please show the whole log. I guess  it's because the permission, please edit the /etc/libvirt/qemu.conf to make qemu run as root user.

**Comment by feiskyer** [comment]
I run with -v=7

File: hypervisor/libvirt/libvirt.go [hunk 3]
```diff
@@ -0,0 +1,699 @@
...
+
+	data, err := xml.Marshal(n)
+	if err != nil {
+		return "", err
+	}
+	return string(data), nil
+}
+
+func (lc *LibvirtContext) AddNic(ctx *hypervisor.VmContext, host *hypervisor.HostNicInfo, guest *hypervisor.GuestNicInfo) {
+	nicXml, err := nicXml(host.Bridge, host.Device, host.Mac, guest.Busaddr)
```
Comment (feiskyer) [inline hypervisor/libvirt/libvirt.go]: `host.Device` should be `guest.Device`

Comment (gao-feng) [inline hypervisor/libvirt/libvirt.go]: Nice catch! updated, thanks

Comment (feiskyer) [inline hypervisor/libvirt/libvirt.go]: Sorry for taking you into the trap, here should be `host.Device` indeed. Network nic will be set up correctly after merging #95 and reverting this change.

**Comment by feiskyer** [comment]
LGTM

**Comment by laijs** [comment]
Good.

**Pull request merged**